src/controllers/legacycontrollersettings.h

```diff
@@ -6,6 +6,16 @@
 #include "controllers/legacycontrollersettingslayout.h"
 #include "util/parented_ptr.h"
 
+namespace {
+template<class T>
+bool valid_range(T min, T max, T step) {
+    VERIFY_OR_DEBUG_ASSERT(step >= 0) {
+        return false;
+    }
```

Comment:
do we actually ensure this somewhere else (and provide appropriate debug feedback)? I'm not particularly familiar with the code.

Reply:
Yes and no. If a setting definition is invalid, you will get a warning log message telling you about it, no actual UI feedback on the matter.

At the time of implementation, the assumption was made that mapping developer would be able to gather log if they needed to

Reply:
No, it just says it will ignore a definition due to an invalid settings

Reply:
thats unfortunate. we should fix that in the future. Just saying something is invalid gets increasingly frustrating the more complex the definition is. how would I know which of my (hypothetical) 10 parameters are invalid? Out-of-scope here though. Just  suggestion.

Reply:
You would know which settings is invalid, but it wouldn't details what's wrong with it (e.g Default out of range? Min after max?)

The warning for that is [here](https://github.com/mixxxdj/mixxx/blob/cb9a5c20cdd962e095ec409751486443aa05af69/src/controllers/legacycontrollermappingfilehandler.cpp#L299)

Let's say you have 10 setitings (`my setting1`..`mySetting10`), and `mySetting6` is invalid, you would get something like
```
Warning: The parsed setting in file "/home/me/myfile.hid/xml" at line 42 appears to be invalid. It will be ignored.
```

Reply:
yeah, thats not great once a setting has a bunch of parameters (along with non-obvious constraints). Ideally the parser should be able to report validation failures as they happen.

Reply:
I might look into that for the future. Happy to get this bugfix in the meantime?

Reply:
yes.